lib/ecto/association.ex

```diff
@@ -247,12 +250,20 @@ defmodule Ecto.Association do
     values = List.wrap(values)
     query = case {join_to, values} do
       {nil, [single_value]} ->
-        query
-        |> where([{dest, final_bind}], field(dest, ^dest_out_key) == ^single_value)
+        dest_out_key
+        |> Enum.zip(single_value)
+        |> Enum.reduce(query, fn {dest_out_key_field, value}, query ->
+          query
+          |> where([{dest, final_bind}], field(dest, ^dest_out_key_field) == ^value)
+        end)
 
       {nil, values} ->
-        query
-        |> where([{dest, final_bind}], field(dest, ^dest_out_key) in ^values)
+        dest_out_key
+        |> Enum.zip(transpose_values(values))
```

Comment:
Can you please add a code comment explaining why we need `transpose_values`? TY!

Reply:
the function was bogus, actually. I've managed to remove it 😅 thank you for pointing me at it.